data/simpy.gram

```diff
@@ -36,44 +36,50 @@ assert_stmt: 'assert' expression [',' expression]
-        _Py_ImportFrom(((expr_ty) c)->v.Name.id, d, seq_count_dots(b), EXTRA(a, d)) }
-    | e='from' f=('.' | '...')+ 'import' g=import_from_targets { _Py_ImportFrom(NULL, g, seq_count_dots(f), EXTRA(e, g)) }
+        _Py_ImportFrom(((expr_ty) c)->v.Name.id, seq_map_to_alias(p, d), seq_count_dots(b), EXTRA(a, token_type, seq_get_tail(NULL, d), alias_type)) }
+    | e='from' f=('.' | '...')+ 'import' g=import_from_targets {
+        _Py_ImportFrom(NULL, seq_map_to_alias(p, g), seq_count_dots(f), EXTRA(e, token_type, seq_get_tail(NULL, g), alias_type)) }
 import_from_targets[asdl_seq*]:
     | '(' a=import_from_as_names ')' { a }
     | import_from_as_names
-    | a='*' { singleton_seq(p, alias_for_star(p)) }
+    | a='*' { singleton_seq(p, pegen_alias(alias_for_star(p), EXTRA(a, token_type, a, token_type))) }
 import_from_as_names[asdl_seq*]: a=import_from_as_name b=(',' c=import_from_as_name { c })* [','] { seq_insert_in_front(p, a, b) }
-import_from_as_name[alias_ty]:
-    | a=NAME 'as' b=NAME { _Py_alias(((expr_ty) a)->v.Name.id, ((expr_ty) b)->v.Name.id, p->arena) }
-    | a=NAME { _Py_alias(((expr_ty) a)->v.Name.id, NULL, p->arena) }
+import_from_as_name[PegenAlias*]:
+    | a=NAME 'as' b=NAME {
+        pegen_alias(_Py_alias(((expr_ty) a)->v.Name.id, ((expr_ty) b)->v.Name.id, p->arena), EXTRA(a, expr_type, b, expr_type)) }
+    | a=NAME {
+        pegen_alias(_Py_alias(((expr_ty) a)->v.Name.id, NULL, p->arena), EXTRA(a, expr_type, a, expr_type)) }
 dotted_as_names[asdl_seq*]: a=dotted_as_name b=(',' c=dotted_as_name { c })* { seq_insert_in_front(p, a, b) }
-dotted_as_name[alias_ty]:
-    | a=dotted_name 'as' b=NAME { _Py_alias(((expr_ty) a)->v.Name.id, ((expr_ty) b)->v.Name.id, p->arena) }
-    | a=dotted_name { _Py_alias(((expr_ty) a)->v.Name.id, NULL, p->arena) }
+dotted_as_name[PegenAlias*]:
+    | a=dotted_name 'as' b=NAME {
+        pegen_alias(_Py_alias(((expr_ty) a)->v.Name.id, ((expr_ty) b)->v.Name.id, p->arena), EXTRA(a, expr_type, b, expr_type)) }
+    | a=dotted_name {
+        pegen_alias(_Py_alias(((expr_ty) a)->v.Name.id, NULL, p->arena), EXTRA(a, expr_type, a, expr_type)) }
 dotted_name[expr_ty]:
     | a=dotted_name '.' b=NAME { join_names_with_dot(p, a, b) }
     | NAME
 
-if_stmt:
-    | 'if' a=full_expression ':' b=block c=else_block { _Py_If(a, b, c, EXTRA(a, c)) }
-    | 'if' a=full_expression ':' b=block c=elif_stmt { _Py_If(a, b, singleton_seq(p, c), EXTRA(a, c)) }
-    | 'if' a=full_expression ':' b=block { _Py_If(a, b, NULL, EXTRA(a, b)) }
-elif_stmt:
-    | 'elif' a=full_expression ':' b=block c=else_block { _Py_If(a, b, c, EXTRA(a, c)) }
-    | 'elif' a=full_expression ':' b=block c=elif_stmt { _Py_If(a, b, singleton_seq(p, c), EXTRA(a, c)) }
-    | 'elif' a=full_expression ':' b=block { _Py_If(a, b, NULL, EXTRA(a, b)) }
-else_block: 'else' ':' b=block { b }
+if_stmt[stmt_ty]:
+    | a='if' b=full_expression ':' c=block d=else_block { _Py_If(b, c, d, EXTRA(a, token_type, seq_get_tail(NULL, d), stmt_type)) }
```

Comment:
I think it's time to name the variables more meaningfully than `a`, `b`, `c` and `d`. (Note: never name a variable `p`. :-)